frontend/src/pages/AccountSettingPage.tsx

```diff
@@ -1,31 +1,138 @@
-import { useCallback, useState } from "react";
-import { Button } from "@mui/material";
+import { useState } from "react";
+import { AccountCircle, Lock } from "@mui/icons-material";
+import {
+  Divider,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  Paper,
+  Stack,
+  Typography,
+} from "@mui/material";
 
-import { UpdatePasswordModal } from "src/components/modals/UpdatePasswordModal";
+import { UpdatePasswordForm } from "src/components/forms/UpdatePasswordForm";
+import { PasswordRequirement } from "src/components/PasswordRequirement";
 import { useAuth } from "src/contexts/AuthContext";
 
 export const AccountSettingPage = () => {
   const { user } = useAuth();
-  const [isOpen, setIsOpen] = useState<boolean>(false);
-
-  const onOpen = useCallback(() => {
-    setIsOpen(true);
-  }, [setIsOpen]);
+  const [formType, setFormType] = useState<"changename" | "changepassword">(
+    "changename",
+  );
 
-  const onClose = useCallback(() => {
-    setIsOpen(false);
-  }, [setIsOpen]);
+  const handleListItemClick = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
+    formType: "changename" | "changepassword",
+  ) => {
+    setFormType(formType);
+  };
 
   return (
-    <>
-      <UpdatePasswordModal
-        userId={user?.userId || 0}
-        open={isOpen}
-        onClose={onClose}
-      />
-      <Button variant="contained" onClick={onOpen}>
-        Update Password
-      </Button>
-    </>
+    <Paper elevation={1} sx={{ borderRadius: 3, mx: 6 }}>
+      <Stack direction="row">
+        <List>
+          <ListItem disablePadding sx={{ bgcolor: "white" }}>
+            <ListItemButton
```

Comment:
Is there a way to refactor this part of the code using NavigationButton? I want to replace this part but the onClick action is inside the NavigationButton. I will need to pass down props if I want to reuse the component here